plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+  
+    if (loadCellX->size() == 0 || loadCellY->size() == 0 || loadCellZ->size() == 0)
+    {
+        yError() << "GazeboYarpContactPressureSensorsArrayDriver: Error parsing parameters: \"loadCellX , ..Y, ..Z\" should be followed by  list";
+        return false;    
+    }
+  
+    if (loadCellX->size() != loadCellNames->size() || loadCellY->size() != loadCellNames->size() || loadCellZ->size() != loadCellNames->size())
+    {
+        yError() << "GazeboYarpContactPressureSensorsArrayDriver: Error parsing parameters: \"loadCellX , ..Y, ..Z\" should be the same size as \"loadCellNames\"";
+        return false;
+    }
+  
+    for (size_t i = 0; i < loadCellNames->size(); i++)
+    {
+        yarp::sig::Vector loadCellLoc(3);
+        loadCellLoc.clear();
+        loadCellLoc.push_back(loadCellX->get(i).asDouble());
+        loadCellLoc.push_back(loadCellY->get(i).asDouble());
+        loadCellLoc.push_back(loadCellZ->get(i).asDouble());
+        this->m_loadCellLocations.push_back(loadCellLoc);
+     }
+  
+     m_contactNormalForces.resize(m_loadCellLocations.size());  
+     if (!this->prepareMappingMatrix())
+     {
+         return false;
+     }
+    
+    return true;
+}
+
+bool GazeboYarpContactPressureSensorsArrayDriver::prepareMappingMatrix()
+{
+    // get the size of the load cell locations
+    int n_cells = this->m_loadCellLocations.size();
+  
+    // Consider w = Af ; where w is the wrench of size 3x1 containing normal force, tangential and radial torques
+    // A is the matrix that maps the array of pressure sensor normal forces f to an resultant wrench w acting 
+    // at the contact. However, we ae solving the inverse problem of mapping a wrench acting at the contact link
+    // to the presure sensor array normal forces, hence we need to obtain the pseudo inverse through
+    // f = (A+)w ; where (A+) is the pseudo inverse. Mapping an equivalent wrench to a number of pressure sensor 
+    // normal forces has infinite solutions, but considering the pseudo inverse gives us the solution with minimized norm.
+    // Hence it is not necessary that the array of pressure sensors simulate the same forces at corresponding locations
+    // similar to the real robot. However, the resultant wrench from the pressure sensor array of normal forces should be valid.
+    // The matrix A is the function of loadCell locations (x, y) with respect to the link coordinate frame
+    yarp::sig::Matrix A; 
+    A.resize(N_AXIS ,n_cells);
+  
+    // Add the respective columns to the mapping matrix
+    for (int i = 0; i < n_cells; i++)
+    {
+        yarp::sig::Vector columnMap;
+        columnMap.clear();
+        columnMap.push_back(1);
+        columnMap.push_back(this->m_loadCellLocations[i][1]); // r_y : y-intercept of the contact point with respect to link origin
+        columnMap.push_back(-this->m_loadCellLocations[i][0]); // -r_x : x-intercept of the contact point with respect to link origin
```

Comment:
I think here we are assuming that the load cells are distributed over a plane perpendicular to the z-axis of the link frame, and that the load cells are measuring the force over the z direction. This assumptions should be documented in the plugin docs.